### PR TITLE
refactor(*):changed class based to function based

### LINKED ISSUE #45: Refactoring Class based components to Functional Based
Working with functional based components is much easier as compared to working with class based components.

### PR DESCRIPTION
**Pull request by arteevraina**
Fixes: #45 #83 
Earlier PR's were closed because some folders were changed in the recent merge.
Please Review @DianaLease @dselman  :)

Signed-off-by: Arteev Raina <arteevraina@gmail.com>

### CODE REVIEW

**Comment by arteevraina** [comment]
@DianaLease I have resolved the merge conflicts in this PR. This PR is ready for review :)

File: src/TemplateEditor/NewTemplateDialog.js [hunk 1]
```diff
@@ -16,42 +16,36 @@ import Typography from 'material-ui/Typography';
...
-                this.setState( {selectedText: asyncResult.value});
+                setSelectedText(asyncResult.value);
             }
         });
   };
 
-  handleCancel = () => {
-      this.setState({ open: false });
+  const handleCancel = () => {
+      setIsOpen(isOpen);
```
Comment (DianaLease) [inline src/TemplateEditor/NewTemplateDialog.js]: this should be `setIsOpen(false)`

File: src/TemplateEditor/NewTemplateDialog.js [hunk 2]
```diff
@@ -16,42 +16,36 @@ import Typography from 'material-ui/Typography';
...
   };
 
-  handleOk = () => {
-      this.setState({ open: false });
+  const handleOk = () => {
+      setIsOpen(isOpen);
```
Comment (DianaLease) [inline src/TemplateEditor/NewTemplateDialog.js]: this should also be `setIsOpen(false)`

File: src/ClauseEditor/NewSmartClauseDialog.js [hunk 3]
```diff
@@ -19,75 +19,66 @@ import { Form, Field } from 'react-final-form';
...
-            open: false,
-            selectedText: '',
-        };
-    }
-
-  handleClickOpen = () => {
+  const handleClickOpen = () => {
     const document = window.Office.context.document;
-    this.setState({ open: true });
+    setIsOpen(!isOpen);
```
Comment (DianaLease) [inline src/ClauseEditor/NewSmartClauseDialog.js]: why was the logic changed here? I think we should leave it as `setIsOpen(true);`

File: src/TemplateEditor/NewTemplateDialog.js [hunk 4]
```diff
@@ -16,42 +16,36 @@ import Typography from 'material-ui/Typography';
...
-            open: false,
-            selectedText: '',
-            templateId: ''
-        };
-    }
-
-  handleClickOpen = () => {
-    this.setState({ open: true });
+  const handleClickOpen = () => {
+    setIsOpen(!isOpen);
```
Comment (DianaLease) [inline src/TemplateEditor/NewTemplateDialog.js]: Here too let's do `setIsOpen(true);`

**Review by DianaLease — CHANGES_REQUESTED** [review]

File: src/ClauseEditor/NewSmartClauseDialog.js (hunk 3 above)
Comment (arteevraina) [reply src/ClauseEditor/NewSmartClauseDialog.js]: @DianaLease I think the variable `isOpen` has been assigned a value `false` .  Therefore, can't I use `!isOpen` as its value is now true?

Comment (DianaLease) [reply src/ClauseEditor/NewSmartClauseDialog.js]: That will work, but being explicit here makes more sense. When `handleClickOpen` is run, we always want that function to set `isOpen` to `true`, not just the negation of the current value of `isOpen` (even if we always think that current value will be `false` when this runs). Does that make sense?

Comment (arteevraina) [reply src/ClauseEditor/NewSmartClauseDialog.js]: Yes, I will make these changes :)

**Comment by arteevraina** [comment]
@DianaLease This PR is ready for Review :)

**Review by DianaLease — APPROVED** [review]
LGTM

**Pull request merged**